examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1468 @@
+        claim_eq!(logger.logs.len(), 1, "One event should be logged");
+        claim_eq!(
+            logger.logs[0],
+            to_bytes(&Cis2Event::<ContractTokenId, ContractTokenAmount>::UpdateOperator(
+                UpdateOperatorEvent {
+                    owner:    ADDRESS_0,
+                    operator: ADDRESS_1,
+                    update:   OperatorUpdate::Add,
+                }
+            )),
+            "Incorrect event emitted"
+        )
+    }
+
+    // test royalties are initialised properly
+    #[concordium_test]
+    fn test_royalties_setup() {
+        // Setup the context
+        let ctx = TestReceiveContext::empty();
+        let mut builder1 = TestStateBuilder::new();
+        let state1 = initial_state(&mut builder1);
+        let host = TestHost::new(state1, builder1);
+        let royalty_paid = supports_interface(&ctx, &host).expect("Invoke should succeed");
+        claim_eq!(royalty_paid, false, "Royalty incorrectly initialised");
+
+        let mut builder2 = TestStateBuilder::new();
+        let mut state2 = initial_state(&mut builder2);
+        state2.pay_royalty = true;
+        let host = TestHost::new(state2, builder2);
+        let royalty_paid = supports_interface(&ctx, &host);
+        claim_eq!(royalty_paid.unwrap(), true, "Royalty incorrectly initialised");
+    }
+
+    // test royalties are initialised properly
+    #[concordium_test]
+    fn test_royalties() {
+        let ctx = TestReceiveContext::empty();
+        let mut builder = TestStateBuilder::new();
+        let state = initial_state(&mut builder);
+        let host = TestHost::new(state, builder);
+        let royalty_paid = supports_interface(&ctx, &host);
+        claim_eq!(royalty_paid.unwrap(), false, "Royalty incorrectly initialised");
+    }
+
+    // test royalties are paid correctly
+    #[concordium_test]
+    fn test_royalties_paid() {
+        // Setup the context
+        let mut ctx = TestReceiveContext::empty();
+        ctx.set_sender(ADDRESS_0);
+
+        // and parameter.
+        let transfer = Transfer {
+            token_id: TOKEN_0,
+            amount:   ContractTokenAmount::from(100),
+            from:     ADDRESS_0,
+            to:       Receiver::from_account(ACCOUNT_1),
+            data:     AdditionalData::empty(),
+        };
+        let parameter = TransferParams::from(vec![transfer]);
+        let parameter_bytes = to_bytes(&parameter);
+        ctx.set_parameter(&parameter_bytes);
+
+        let mut logger = TestLogger::init();
+        let mut state_builder = TestStateBuilder::new();
+        let state = initial_state_with_royalties(&mut state_builder);
+        let mut host = TestHost::new(state, state_builder);
+
+        // Call the contract function.
+        let result: ContractResult<()> = contract_transfer(&ctx, &mut host, &mut logger);
+        // Check the result.
+        claim!(result.is_ok(), "Results in rejection");
+
+        // Check the state.
+        let balance0 =
```

Comment:
Can you check the balance of the three entities `royalty_receiver`, `from_address`, and ` to_address` and use different addresses for all the three, please.